File: quarkus-karate-optiker/src/main/java/at/htl/karate/boundary/UserEndpoint.java
package at.htl.karate.boundary;

import at.htl.karate.control.UserRepository;
import at.htl.karate.entity.User;
import io.quarkus.hibernate.orm.panache.PanacheQuery;
import javax.inject.Inject;
import javax.transaction.Transactional;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;

@Path("/user")
public class UserEndpoint {
    @Inject
    UserRepository userRepository;

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public PanacheQuery<User> findAll() {
        return userRepository.findAll();
    }

    @POST
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    @Transactional
    public User create(User user) {
        return userRepository.saveUser(user);
    }

    @PUT
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    @Transactional
    public User update(User user) {
        return userRepository.updateUser(user);
    }
    @DELETE
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    @Transactional
    public User delete(User user) {
        return userRepository.deleteUser(user.id);
    }

    @DELETE
    @Path("/{id}")
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    @Transactional
    public User delete(@PathParam("id") long id, User user) {
        user.id = id;
        return userRepository.deleteUser(user.id);
    }
}

File: quarkus-karate-optiker/src/main/java/at/htl/karate/boundary/UserEndpoint.kt
package at.htl.karate.boundary

import at.htl.karate.control.UserRepository
import at.htl.karate.entity.User
import org.hibernate.PersistentObjectException
import javax.inject.Inject
import javax.persistence.PersistenceContext
import javax.persistence.PersistenceException
import javax.transaction.Transactional
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/user")
@Produces(MediaType.APPLICATION_JSON)
class UserEndpoint {

    @Inject
    lateinit var userDao: UserRepository

    @GET
    fun findAll(): List<User>{
        return userDao.findAll();
    }

    @GET
    @Path("{id}")
    fun getById(@PathParam("id") id: Long): Response {
        val quote = quoteDao.findById(id) ?: return Response.noContent().build()
        return Response.ok(quote).build()
    }

    @POST
    @Transactional
    fun create(user: User): Response {
        userDao.save(user)
        return Response.ok(user).build()
    }


    @PUT
    fun update(user: User): User {
        return userRepository.updateUser(user)
    }

    @PUT
    @Transactional
    @Path("{id}")
    fun update(@PathParam("id") id: Long, quote: Quote): Quote {
        quote.id = id
        quoteDao.mergeAndFlush(quote)
        quoteDao.refresh(quote)
        return quote
    }

    @DELETE
    @Transactional
    fun delete(quote: Quote): Quote {
        quoteDao.delete(quote)
        return quote
    }

    @DELETE
    @Path("{id}")
    @Transactional
    fun delete(@PathParam("id") id: Long): Quote {
        val quote = quoteDao.findById(id);
        quoteDao.delete(quote)
        return quote
    }
}
